### PR TITLE
[WIP] Add new proto file to control what we expose via json api

### PR DESCRIPTION
**Pull request by nakhbari**
This should most likely be a placeholder for whatever protos we use in the future. But the idea here is that we separate the models we expose from the ones we pass around in the backend.

### CODE REVIEW

**Comment by taquitos** [comment]
As mentioned in slack:

> Heya, I think it would really help out if we put together a design doc around the proto work that is going on so that we all can have the same shared context around what problems we’re solving with protos and how we’re thinking about architecting it. That will also help to share knowledge for people who aren’t familiar with this stuff in the future :thumbsup:

File: app/features-json/proto/project_summary_proto.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+module FastlaneCI
+  # Proto to expose the basic info about a project.
+  class ProjectSummaryProto
+    class << self
+      def object_class
+        FastlaneCI::Project
+      end
+
+      def proto_from(object:)
```
Comment (taquitos) [inline app/features-json/proto/project_summary_proto.rb]: I don't know that `proto` is the right word for this. I understand you're thinking about protobufs, but maybe there is some other word we can use that is more familiar to devs who don't know protobuf stuff.

Comment (nakhbari) [reply app/features-json/proto/project_summary_proto.rb]: I 100% agree. I don't know what the correct term is. But this is something much simpler and might eventually get replaced by something more like a proto.

Comment (taquitos) [reply app/features-json/proto/project_summary_proto.rb]: If this is used solely for the view, I might learn towards `ViewData`

**Comment by minuscorp** [comment]
I'll try to work on the idea that created this PR from tomorrow and see if that can make the trick for you!

File: app/features-json/proto/project_summary_proto.rb (hunk 1 above)
Comment (minuscorp) [reply app/features-json/proto/project_summary_proto.rb]: `ViewModel`? 😇

**Comment by minuscorp** [comment]
I'll try to get this thing in a PR tomorrow, I was a little busy today 😅

**Pull request closed without merging**